enhancements/dns/aro-private-dns-zone-resource-removal.md

```diff
@@ -0,0 +1,180 @@
+their vnet configuration.
+
+A working PoC-quality implementation of this approach can be found at
+https://github.com/Azure/ARO-RP/pull/1296.  This implementation is notably
+similar to the following [Machine Config Operator
+code](https://github.com/openshift/machine-config-operator/blob/master/templates/common/on-prem/files/coredns-corefile.yaml),
+however by using dnsmasq which is already present in the RHCOS image, it does
+not depend on the ability to pull container images.  This will be valuable when
+implementing zero Internet egress.
+
+We do not believe that the PoC implementation is trivially applicable to the
+OpenShift installer because it is dependent on calculating the IP addresses for
+{api,api-int,\*.apps}.*cluster domain* before the installer graph is created.
+We suppose this might imply larger installer architecture changes in the
+OpenShift installer IPI general case were this ever to be implemented there, but
+it is relatively straightforward in ARO.
+
+We do believe that the architectural changes/constraints placed on OCP could be
+feasible.  These are:
+
+* ARO clusters would no longer have associated Azure private DNS zones.
+
+* Dnsmasq would be used as an inline proxying DNS resolver for all ARO cluster
+  node-context and pod-context DNS resolution.
+
+* Dnsmasq would need to continue to be provided in the RHCOS image.
+
+* ARO VMs would no longer be DNS resolvable by their hostname from other VMs.
+  Note that because --kubelet-preferred-address-types is set to InternalIP in
+  the Kubelet config, we don't believe this is currently a hard platform
+  requirement.
+
+* ARO machine config server URL and serving certificate would use api-int IP
+  address instead of DNS.  This is because DNS resolution of api-int would not
+  be available at ignition time.
+
+* The ARO cluster installation process would prepopulate the
+  openshift-ingress/router-default LoadBalancer service early with the
+  appropriate IP address.
+
+* Ingress operator dynamic DNS would be disabled on ARO via overridden
+  configuration of the DNS custom resource.
+
+### Risks and Mitigations
+
+One downside of the proposed solution is that it makes it somewhat harder,
+although not infeasible, to handle the case of cluster recovery if the api-int
+IP address changes.  This is not expected to be a likely occurrence.  ARO
+already protects against this case by preventing customers from manually
+modifying the configuration of the Azure cluster load balancer resources.  ARO
+would be responsible for automatically handling *.apps IP address changes (also
+unlikely) as well as defining a process to recover from api-int IP address
+changes if ever needed.
+
+Removing the ability to resolve VM hostnames via DNS could conceivably impact
+end user workloads, but the likelihood of this is considered low.  We don't
+consider the ability to resolve VM hostnames via DNS part of the ARO platform
```

Comment:
what about oc logs / oc debug commands - will they work without hosts being resolvable?

Reply:
I checked oc logs, debug, and rsh in my local env without node names being resolvable and they all worked fine.